File: 백준/Bronze/10798. 세로읽기/세로읽기.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;

public class Main {
    public static void main(String[] args) throws IOException {

        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringBuilder sb = new StringBuilder();
        
        char[][] map = new char[5][];
        int maxlen = 0;
        for (int i=0; i<5; i++) {
        	map[i] = br.readLine().toCharArray();
        	maxlen = Math.max(maxlen, map[i].length);
        }
        
        for (int i=0; i<maxlen; i++) {
        	for (int j=0; j<5; j++) {
        		if (i < map[j].length) sb.append(map[j][i]);
        	}
        }
        
        System.out.println(sb.toString());
        
    }
}

File: 백준/Bronze/10798. 세로읽기/세로읽기.py
import sys

strings = []
maxLen = 0
for _ in range(5) :
    s = sys.stdin.readline().rstrip()
    maxLen = max(maxLen, len(s))
    strings.append(s)
output = []
for i in range(maxLen) :
    for j in range(5) :
        if i < len(strings[j]) :
            output.append(strings[j][i])
print("".join(output))